### DISCUSSION TITLE
Do tree-sitter queries _always_ start with `(` aka opening parenthesis?

### DISCUSSION THREAD

**Discussion by alexpovel** [tree-sitter/tree-sitter#3862] [Q&A] [answered]
Hey folks!

Over at https://github.com/alexpovel/srgn/pull/144 , we are building a feature for a tree-sitter based tool. The feature is a CLI flag accepting a string, which is _either_ a normal tree-sitter query, _or_ a file path to a file containing such a query: if you pass `(some_query)`, it will be taken as a tree-sitter query; if you pass `some/file.scm`, the query will be read from that file.

To disambiguate, it would be very helpful to know with certainty **whether tree-sitter queries aka S-expressions always have to start with `(`**?

From experience writing them, and from the [docs](https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax), this seems to be the case, but I was hoping to find a more authoritative answer here in the community 🙂 If the answer is _no_, is there at least a set of well-known, obligatory first characters (something like "`(`, `[`, ...")?

**Comment by clason** [comment, marked as answer]
No, the following is a perfectly valid pattern (and hence minimal query):

```
"null" @constant.builtin
```

Also valid would be
```
[
  "true"
  "false"
] @boolean
```
(No parens anywhere! It's hardly a Lisp...)

You can find a huge corpus for playing around in https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries.

**Comment by clason** [comment]
Honestly, I think having polyvalent flags like that is not good design; in fact, I don't see good reason for not simply having two flags, or the standard `-` "read from `stdin`".
